applications/cheap_flights/flights_search.py

```diff
@@ -67,13 +66,25 @@ def search_flights(**kwargs):
     * Move some data into a variable and do a filter from there
     * Keep adding IF (like current one)
     * Quit dev(LOL)
+    
+    TOANSWER: 
+    Add your "escape hatches" early and organize the if statements in a smart way.  If it's code 400, return.  
+    There is no point even looking at the rest of the code.  If you do it in this order, the last `else` isn't 
+    needed at all because each of the other cases `return` so the only way to get there now is if neither of 
+    the other if statements are true. 
     """
+    if response.status_code == 400:
+        return result, response.status_code
+
     if response.status_code == 200:
         for data in result["data"]:
-            if data["availability"]["seats"] is not None and data["availability"]["seats"] >= adults:
```

Comment:
Python is actually pretty smart.  You have
```if data["availability"]["seats"] is not None``` 

but if you just say 

```if data["availability"]["seats"]```

Python reads that as 

```if data["availability"]["seats"] is not None and data["availability"]["seats"] is not False```

Reply:
When i was doing the tests it was letting it pass, i guess bc it was None there

I would have to go back and test again

I'll update there

Reply:
ok, just tested and it worked .... not sure what to say lol

edited as recommended.

Reply:
Just keep it in mind also, it's really handy but sometimes it bites you :stuck_out_tongue: